README.md

```diff
@@ -15,58 +15,146 @@ The library ships with a number of features:
-```
+Taxonomy can be loaded from a variety of sources.
+
+1. `Taxonomy.from_newick(value: str)`: loads a Taxonomy from a Newick-encoded string.
+
+2. `Taxonomy.from_ncbi(nodes_path: str, names_path: str)`: loads a Taxonomy from a pair of NCBI dump files. The paths specified are
+to the individual files in the NCBI taxonomy directory (e.g. nodes.dmp and names.dmp).
+
+3. `Taxonomy.from_json(value: str, /, path: List[str])`: loads a Taxonomy from a JSON-encoded string. The format can either be
+of the tree or node_link_data types and will be automatically detected. If `path` is specified, the JSON will be traversed to that sub-object before being parsed as a taxonomy.
+
+4. `Taxonomy.from_phyloxml(value: &str)`: loads a Taxonomy from a PhyloXML-encoded string. **Experimental**
+
+### Exporting a taxonomy
+
+Assuming that the taxonomy has been instantiated as a variable named `tax`.
+
+1. `tax.to_newick()`: exports a Taxonomy as a Newick-encoded byte string.
 
-If you have the NCBI taxonomy locally ([found on their FTP](ftp://ftp.ncbi.nlm.nih.gov/pub/taxonomy/taxdump.tar.gz)), you can use that too:
+2. `tax.to_json(/, as_node_link: bool)`: exports a Taxonomy as a JSON-encoded byte string. By default, the JSON format 
+is a tree format unless the `as_node_link` parameter is set to True.
+
+### Using a taxonomy
+
+Assuming that the taxonomy has been instantiated as a variable named `tax`. Note that `TaxonomyNode` is a class with
+the following schema:
 
 ```python
-ncbi_tax = Taxonomy.from_ncbi('./nodes.dmp', './names.dmp')
-assert tax.name('562') == 'Escherichia coli'
-assert tax.rank('562') == 'species'
+class TaxonomyNode:
+    id: str
+    name: str
+    parent: Optional[str]
+    rank: str
 ```
 
-Note that Taxonomy IDs in NCBI format are integers, but they're converted to strings on import. We find working with "string taxonomy IDs" greatly simplifies interoperation between different taxonomy systems.
+Note that tax_id in parameters passed in functions described below are string but for example in the case of NCBI need
+to be essentially quoting integers: `562 -> "562"`. In that case, passing something that can't be converted to a number
+will raise an exception even if the documentation below does not mention it.
+
+#### `tax.root -> TaxonomyNode`
+Points to the root of the taxonomy
+
+#### `tax.parent(tax_id: str, /, at_rank: str) -> (Optional[TaxonomyNode], Optional[float])`
+Return the immediate parent TaxonomyNode of the node id provided as well as the distance to it as a `(TaxonomyNode, float)` tuple.
+
+If `at_rank` is provided, scan all the nodes in the node's lineage and return
+the parent id at that rank.
+
+Examples:
+
+```py
+parent, distance = tax.parent("612")
+parent, _ = tax.parent("612", at_rank="species")
+parent, distance = tax.parent("612")
+# Both variables will be `None` if we can't find the parent
+parent, distance = tax.parent("unknown")
+```
+
+#### `tax.find_by(tax_id: str) -> Optional[TaxonomyNode]`
```

Comment:
I'd prefer `.get()` for consistency with Python dict syntax. I'd expect a `find_by` method to do some magic where I can pass a string and get a list of matching tax IDs back.